crates/ursa-rpc-service/src/api.rs

```diff
@@ -216,39 +386,8 @@ where
         let size = car.size;
         let cids = load_car(self.store.blockstore(), car).await?;
         let root_cid = cids[0];
-
         info!("The inserted cids are: {cids:?}");
-
-        let (sender, receiver) = oneshot::channel();
-        let request = NetworkCommand::Put {
-            cid: root_cid,
-            sender,
-        };
-        if let Err(e) = self.network_send.send(request) {
-            error!("There was an error while sending NetworkCommand::Put: {e}");
-        } else if let Err(e) = receiver.await {
-            return Err(anyhow!(format!(
-                "The PUT failed, please check server logs {e:?}"
-            )));
-        }
-
-        let (sender, receiver) = oneshot::channel();
-        let request = ProviderCommand::Put {
-            context_id: root_cid.to_bytes(),
-            size,
-            sender,
-        };
-        if let Err(e) = self.provider_send.send(request) {
-            // this error can be ignored for test_put_and_get test case
-            error!("there was an error while sending provider command {e}");
-            return Ok(cids);
-        }
-        match receiver.await {
-            Ok(_) => Ok(cids),
-            Err(e) => Err(anyhow!(format!(
-                "The PUT failed, please check server logs {e:?}"
-            ))),
-        }
+        self.provide_cid(root_cid, size).await?.map(|_| cids)
```

Comment:
here we spawn a nested task. We should prob revert this and just use `put_car`.

Reply:
yes and immediately awaited, just a layer of indirection. need the spawn so that network and provider notification is non blocking, so that get/get_data responds without extra delay

Reply:
but it is not, it does block here and has to wait for the task. We have the server task which handles each of the commands separately, then you are spawning another task.

Reply:
https://docs.rs/axum/latest/axum/ for reference

Reply:
This is not the cache miss handler so it shouldn't care about that function

Reply:
you should be able to spawn this "blocking" task

Reply:
https://github.com/fleek-network/ursa/pull/278/commits/fbc8f99c5acc0fce42e47de0f407faea4f53ff37